pkg/tracing/tracer.go

```diff
@@ -224,6 +224,10 @@ func (t *Tracer) jobProcessor(ctx context.Context, jobChan <-chan *Job) {
 
 // AddJob add a job to tracer
 func (t *Tracer) AddJob(job *Job) {
+	if t.jobsClosed.Get() {
```

Comment:
does it really resolve data race problem?

Reply:
yep, if we want to avoid race condition without using `Lock`, we must ensure initial Tracer before any job operation

Reply:
So we only set `jobsClosed` to false at the end of `newJobChans`, which can guarantee no access to jobs channels before they are created.

Reply:
there are still temporal logic conflict problem

Reply:
yep, it is not good design, I add lock to protect the access to jobs channel